### PR TITLE
Wiki-enu-flu-conversion

### PR DESCRIPTION
**Pull request by joachimhgg**
**Description**
Wiki of the ENU/FLU conversion.
This PR is not intented to be merge as the doc will be added to the wiki. It is just to make it easier to review it. Images and links (to the scripts, etc) will be added in the wiki.

### CODE REVIEW

**Comment by lukicdarkoo** [comment]
I believe NUE -> ENU is not relevant for users, but FLU. The FLU conversion will break a lot of stuff (not NUE -> ENU) and we should focus on that.

File: docs/wiki_flu.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# New axis system for geometries, devices, and PROTOs
+
+Starting from the R2022a version, all geometries, devices, and PROTOs that come with Webots use the FLU axis orientation (x-**F**orward, y-**L**eft, and z-**U**p, see [#3073](https://github.com/cyberbotics/webots/pull/3073)).
+We introduced the update to be consistent with ROS (see [REP 103](https://www.ros.org/reps/rep-0103.html)) and other robotics systems.
+If your Webots world uses Webots geometries, devices, and PROTOs then you will probably need to adapt your world.
+
+[add image example of a 2021b world with the axis and a 2022a world with the axis]
```
Comment (omichel) [inline docs/wiki_flu.md:7]: It would be nice to actually add this image now.

File: docs/wiki_flu.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+# How to adapt your world or PROTO to Webots R2022a
+
+## Backward compatibility mechanism
+We introduced a backward compatibility mechanism (see [#3619](https://github.com/cyberbotics/webots/pull/3619)) to adapt worlds automatically. The mechanism will be launch when you open a world in the R2022a version.
```
Comment (omichel) [inline docs/wiki_flu.md:12]:
```suggestion
We introduced a backward compatibility mechanism (see [#3619](https://github.com/cyberbotics/webots/pull/3619)) to adapt worlds automatically.
This mechanism will be launched when you open a world in the R2022a version.
```

File: docs/wiki_flu.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+| [Pen](reference/camera.md) | (-$\frac{\pi}{2}$, 0, 0) | A2 |
+| [Emitter](reference/emitter.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| [Receiver](reference/receiver.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| [Connector](reference/connector.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| [TouchSensor](reference/touchsensor.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| Webots PROTOs | (-$\frac{\pi}{2}$, 0, $\frac{\pi}{2}$), but there are numerous exceptions to this rule as we haven't followed any specific convention | C |
+
+## Automatic conversion to ENU/FLU
+
+This script `scripts/converter/convert_nue_to_enu_rub_to_flu.py` indents to help you to adapt your world or PROTOs from an old version to the new version of Webots.
```
Comment (omichel) [inline docs/wiki_flu.md:42]: We should add a hyperlink to the script in the develop branch.
```suggestion
This script `scripts/converter/convert_nue_to_enu_rub_to_flu.py` intends to help you to adapt your world or PROTOs from an old version to the new version of Webots.
```

File: docs/wiki_flu.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+[add image example of a "broken" NUE (2021b) to ENU (R2022a) complex proto]
+
+This script will convert your world from _NUE_ to _ENU_ and the objects contained in the world from _RUB_ to _FLU_. It could be needed to rotate some parts by hand. 
+
+Simple usage:
+```
+python3 convert_nue_to_enu_rub_to_flu.py /your_path_to_your_projects/worlds/my_world.wbt
+```
+
+If your PROTO is _RUB_ (x-**R**ight, y-**U**p, z-**B**ack) and does not contain JavaScript or Lua code, you can convert it to _FLU_ using the same script. You may have to rotate some parts of your PROTOs by hand.
```
Comment (omichel) [inline docs/wiki_flu.md:55]:
```suggestion
If your PROTO is _RUB_ (x-**R**ight, y-**U**p, z-**B**ack) and does not contain any JavaScript or Lua code, you can convert it to _FLU_ using the same script. You may have to rotate some parts of your PROTOs manually.
```

File: docs/wiki_flu.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+**Notes:** check that the conversion went successfully by displaying the different rendering `View/Optional rendering`. In addition, we advise you to check the differences of your `.wbt` or `.proto` before and after the script in a text comparator.
+# New axis system recommendations
+
+As of Webots R2022a, we recommend the FLU (x-**F**orward, y-**L**eft, and z-**U**p) axis orientation for objects and ENU (x-**E**ast, y-**N**orth, and z-**U**p, see [Axes conventions
+](https://en.wikipedia.org/wiki/Axes_conventions)) for worlds.
```
Comment (omichel) [inline docs/wiki_flu.md:61]:
```suggestion
As of Webots R2022a, we recommend the FLU (x-**F**orward, y-**L**eft, and z-**U**p) axis orientation for objects and ENU (x-**E**ast, y-**N**orth, and z-**U**p, see [Axes conventions](https://en.wikipedia.org/wiki/Axes_conventions)) for worlds.
```

**Review by omichel — CHANGES_REQUESTED** [review]

File: docs/wiki_flu.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+| [Emitter](reference/emitter.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| [Receiver](reference/receiver.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| [Connector](reference/connector.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| [TouchSensor](reference/touchsensor.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| Webots PROTOs | (-$\frac{\pi}{2}$, 0, $\frac{\pi}{2}$), but there are numerous exceptions to this rule as we haven't followed any specific convention | C |
+
+## Automatic conversion to ENU/FLU
+
+This script `scripts/converter/convert_nue_to_enu_rub_to_flu.py` indents to help you to adapt your world or PROTOs from an old version to the new version of Webots.
+You can find detailed explication into the script (dependencies, usage, limitations and conversion process).
```
[the remark this answers is not in the document]
Comment (omichel) [reply docs/wiki_flu.md:44]: This was not addressed.

File: docs/wiki_flu.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+[add image example of a "broken" RUB (2021b) to FLU (R2022a) complex proto]
+[add image example of a "broken" NUE (2021b) to ENU (R2022a) complex proto]
+
+This script will convert your world from _NUE_ to _ENU_ and the objects contained in the world from _RUB_ to _FLU_. It could be needed to rotate some parts by hand.
```
[the remark this answers is not in the document]
Comment (omichel) [reply docs/wiki_flu.md:49]: This was not addressed.

**Review by omichel — CHANGES_REQUESTED** [review]
Several of my requests were not taken into account.

File: docs/wiki_flu.md (hunk 1 above)
Comment (joachimhgg) [reply docs/wiki_flu.md:7]: spot2
Example of spot.wbt in 2021b: spot is RUB and the world is NUE.
spot1
Example of spot.wbt in 2022a: spot is FLU and the world is ENU.

File: docs/wiki_flu.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+| [TouchSensor](reference/touchsensor.md) | (-$\frac{\pi}{2}$, 0, -$\frac{\pi}{2}$) | A2 |
+| Webots PROTOs | (-$\frac{\pi}{2}$, 0, $\frac{\pi}{2}$), but there are numerous exceptions to this rule as we haven't followed any specific convention | C |
+
+## Automatic conversion to ENU/FLU
+
+This script `scripts/converter/convert_nue_to_enu_rub_to_flu.py` indents to help you to adapt your world or PROTOs from an old version to the new version of Webots.
+You can find detailed explication into the script (dependencies, usage, limitations and conversion process).
+
+[add image example of a "broken" RUB (2021b) to FLU (R2022a) complex proto]
+[add image example of a "broken" NUE (2021b) to ENU (R2022a) complex proto]
```
[the remark this answers is not in the document]
Comment (joachimhgg) [reply docs/wiki_flu.md:47]: vil1
Example of a complex world after backward compatibility algorithm.
vil2
Example of a complex world after the execution of the script.

[the remark this answers is not in the document]
Comment (joachimhgg) [reply docs/wiki_flu.md:47]: other proposition:
highway
highway2

city
156.png)
city2
c2fd

hall
hall2

complete_appartment1
complete_appartment2

[the remark this answers is not in the document]
Comment (omichel) [reply docs/wiki_flu.md:47]: The first one looks good to me.

**Review by omichel — APPROVED** [review]
Thank you.

**Comment by lukicdarkoo** [comment]
The wiki is not indexed and it is not accessible from the Webots docs. Is there a plan to publish it? Should we put it in the `guide` directory?

**Comment by joachimhgg** [comment]
The wiki has been created : https://github.com/cyberbotics/webots/wiki/FLU-conversion-guide
I am closing the PR.

**Pull request closed without merging**